sycl/source/detail/error_handling/enqueue.cpp

```diff
@@ -0,0 +1,179 @@
+//===-- enqueue.cpp - SYCL error handling of enqueue operations -*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "error_handling.hpp"
+
+#include <CL/sycl/detail/pi.hpp>
+
+namespace cl {
+namespace sycl {
+namespace detail {
+
+namespace enqueue_kernel_launch {
+
+bool handleInvalidWorkGroupSize(pi_device Device, pi_kernel Kernel,
+                                NDRDescT NDRDesc) {
+  bool HasLocalSize = (NDRDesc.LocalSize[0] != 0);
+
+  size_t VerSize;
+  PI_CALL(RT::piDeviceGetInfo, Device, PI_DEVICE_VERSION, 0, nullptr, &VerSize);
+  string_class Ver(VerSize, '\0');
```

Comment:
Is it really required to be string_class and not a simple array of chars?

Reply:
I don't know the size, so, I have to allocate a memory dynamically. I don't want to deal with `new`/`delete` if `string_class` can do this instead of me.